### PR TITLE
rewrite Tag Replacer

### LINKED ISSUE #1289: Tag Replacer's option is confusing/broken
What I expected leaving `Case Sensitive` disabled to do: Change `tag` and `Tag` to `Replacement Tag`

What it actually does: Changes `tag` and `Tag` to `Replacement Tag` *and lowercases all other tags on every modified post*

---

What I expected enabling it to do: Change `Tag` to `Replacement Tag`

What it actually does: Breaks. Definitely fails to find any results if the tag has a space in it, might break in other times

---

Will try and fix later.

### PR DESCRIPTION
**Pull request by AprilSylph**
makes Tag Replacer's code make sense

- resolves #1289 
- removes a ton of single-line XKit.window calls
- puts the functions in the order that theyre used
- did i mention i fixed case sensitive mode
- (and that tags we're not replacing are left completely untouched now)
- (as they should be)

### CODE REVIEW

File: Extensions/tag_replacer.js [hunk 1]
```diff
@@ -44,268 +42,265 @@ XKit.extensions.tag_replacer = new Object({
 
 	show: function(url) {
 
-		XKit.window.show("Tag Replacer", "<b>Replace this tag:</b><input type=\"text\" maxlength=\"150\" placeholder=\"Enter a tag (example: 'I like pandas')\" class=\"xkit-textbox\" id=\"xkit-tag-replacer-replace\"><div class=\"xkit-checkbox\" id=\"xkit-tag-replacer-case-sensitive\"><b>&nbsp;</b>Case Sensitive Mode</div><div class=\"xkit-tag-replacer-separator\">&nbsp;</div><b>With this tag:</b> (leave blank to delete the tag)<input type=\"text\" maxlength=\"150\" placeholder=\"Enter a tag (example: 'I still like pandas')\" class=\"xkit-textbox\" id=\"xkit-tag-replacer-with\"><div class=\"xkit-checkbox\" id=\"xkit-tag-replacer-append\"><b>&nbsp;</b>Don't replace the tag but append the tag above</div><div class=\"xkit-tag-replacer-separator\">&nbsp;</div><small>You can replace only one tag at a time.<br/>Due to technical reasons, you can't edit tags containing dashes.</small>", "question", "<div class=\"xkit-button default\" id=\"xkit-tag-replacer-ok\">Go!</div><div class=\"xkit-button\" id=\"xkit-close-message\">Cancel</div>");
+		XKit.window.show(
+			"Tag Replacer",
```
Comment (hobinjk) [inline Extensions/tag_replacer.js]: nit: Mixing ' and " hurts my soul

though it does cut down on escaping so I guess it's okay

File: Extensions/tag_replacer.js [hunk 2]
```diff
@@ -44,268 +42,265 @@ XKit.extensions.tag_replacer = new Object({
...
 	case_sensitive: false,
 	success_count: 0,
 	fail_count: 0,
 
-	start: function(url, t_replace, t_with, append_mode) {
-
-		console.log("replace " + t_replace + " with " + t_with + " on " + url);
+	start: function(url, replace, replace_with) {
+
+		Object.assign(this, {
```
Comment (hobinjk) [inline Extensions/tag_replacer.js]: Very fancy!

File: Extensions/tag_replacer.js [hunk 3]
```diff
@@ -44,268 +42,265 @@ XKit.extensions.tag_replacer = new Object({
...
+				if (data.response.posts.length === 0) {
+					this.start_replace();
+					return;
+				}
+
+				data.response.posts.forEach(post => this.post_ids.push(post.id));
+				$("#xkit-tag-replacer-loaded").html(`Loaded ${this.post_ids.length} tagged posts...`);
 
-		XKit.window.show("Tag Replacer encountered an error", message, "error", "<div class=\"xkit-button default\" id=\"xkit-close-message\">OK</div>");
+				this.fetch_posts(++page);
```
Comment (hobinjk) [inline Extensions/tag_replacer.js]: This feels unnecessarily terse compared to `page + 1`

Comment (AprilSylph) [reply Extensions/tag_replacer.js]: i like it too much im sorry

Comment (nightpool) [reply Extensions/tag_replacer.js]: agree w/ once, this is officially Too Cute

Comment (AprilSylph) [reply Extensions/tag_replacer.js]: fine :(

Comment (nightpool) [reply Extensions/tag_replacer.js]: (the main problem is that ++ implies you also care about mutating the variable, which you don't)

File: Extensions/tag_replacer.js (hunk 1 above)
Comment (AprilSylph) [reply Extensions/tag_replacer.js]: yeah the only reason im using single-quotes anywhere is to put html without escaping all the attribute quotes and i just inexplicably like double-quotes by default. there's only a few apostrophies i'd have to escape if i used single-quotes more consistently

**Comment by AprilSylph** [comment]
i'm not gonna merge this yet since @nightpool has valid concerns about people seeing Tag Replacer update, assuming we've fixed the tag-replacer-triggers-tumblrs-spam-filter-and-terminates-you thing (#1665), then using it and getting their account terminated.

**Comment by AprilSylph** [comment]
image

**Pull request merged**